control/subscription_group.go

```diff
@@ -405,23 +409,53 @@ func (s *subscriptionGroup) subscribePlugins(id string,
 			"version": plg.Version(),
 			"_block":  "subscriptionGroup.subscribePlugins",
 		}).Debug("plugin subscription")
-		pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
-		if err != nil {
-			serrs = append(serrs, serror.New(err))
-			return serrs
-		}
-		pool.Subscribe(id)
-		if pool.Eligible() {
-			err = s.verifyPlugin(plg)
+		if plg.Details.Uri != nil {
+			// this is a remote plugin
+			pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
```

Comment:
Checking of this err is moved to bottom. Can you move it back to below this line?